### DISCUSSION TITLE
ERR_EMPTY_RESPONSE with firewall open on macOS 15.0

### DISCUSSION THREAD

**Discussion by Intro1997** [navidrome/navidrome#3318] [Q&A] [answered]
I download application from here:
https://github.com/navidrome/navidrome/releases/download/v0.53.2/navidrome_0.53.2_darwin_amd64.tar.gz 
and move application to user folder `~/xxx/navidrome`.
Then I create config file `navidrome_config.toml`:
```
# This is just an example! Please see available options to customize Navidrome for your needs at
# https://www.navidrome.org/docs/usage/configuration-options/#available-options

LogLevel = 'DEBUG'
ScanSchedule = '@every 24h'
TranscodingCacheSize = '200MiB'
MusicFolder = '/Volumes/Samsung_T5/Music/Song'

Address = '192.168.0.194'
port = '20492'

# This is the default path for Homebrew installed ffmpeg
FFmpegPath = '/opt/homebrew/bin/ffmpeg'
```
then use create `start.sh` and write command `./navidrome --configfile "navidrome_config.toml"` to `start.sh`, then use `./start.sh` to start server. 

The output in terminal looks ok, but I when access `192.168.0.194:20492` by Chrome, the page return `ERR_EMPTY_RESPONSE` error (but if I use 127.0.0.1:20492, it works fine).

I had added `navidrome` application to firewall, but I still get error `ERR_EMPTY_RESPONSE`. I also check port listen state by`lsof -c navidrome`, it looks no problem.

When I close firewall, I can connect to the server.

I also try to build application from source code, it works final with address `192.168.0.194:20492`, even if I redownload it from `http-server` which started by myself (A security pop-up window will appear when open it), it still works fine.

I don't want to close my firewall or update `navidrome` by rebuild from source code, what should I do with it?

**Comment by Intro1997** [comment]
My MacBook uses an M3 Max chip. When running the applications on the Github release page, I need to run via rosetta, is this related?

**Reply by metalheim** [reply]
probably. If you build locally to arm64 arch and it works, but it fails when using pre-buildt amd64 on rosetta, then it must be related to rosetta. 

My guess is that rosetta doesn't respect the firewall rule you have created?

**Reply by Intro1997** [reply]
My firewall only configures whether to allow incoming connections for some applications, and there are no other rules:
<img width="827" alt="Screenshot 2024-09-27 at 00 19 40" src="https://github.com/user-attachments/assets/997cfec3-17ad-4908-9877-199be6682aed">
I don't know if there is a finer-grained way to configure the firewall.

**Reply by deluan** [reply, marked as answer]
If the aarch64 works for you, you could install it with Homebrew instead of downloading from our GitHub page. They have a [formula](https://formulae.brew.sh/formula/navidrome) for Navidrome that is auto-updated every time there is a new ND release, and it installs aarch64 on the new Apple Silicon.

I use Homebrew for almost all sw installations on my macOS, but haven't tried the ND formula myself. I read it though, and seems to be solid.

**Reply by Intro1997** [reply]
Oh! I didn't notice that Navidrome is available on Homebrew! I try it and It works great! Thanks!

**Reply by Intro1997** [reply]
> If the aarch64 works for you, you could install it with Homebrew instead of downloading from our GitHub page. They have a [formula](https://formulae.brew.sh/formula/navidrome) for Navidrome that is auto-updated every time there is a new ND release, and it installs aarch64 on the new Apple Silicon.
> 
> I use Homebrew for almost all sw installations on my macOS, but haven't tried the ND formula myself. I read it though, and seems to be solid.

I have looked up homebrew word in following docs about installation of Navidrome:
1. [installation | Navidrome](https://www.navidrome.org/docs/installation/)
2. [macOS Install | Navidrome](https://www.navidrome.org/docs/installation/macos/)
3. [Github README.md](https://github.com/navidrome/navidrome?tab=readme-ov-file)

But I found nothing about homebrew install. 

Is there any documentation on this installation method? If not, would you consider adding this installation method to the documentation?

**Reply by deluan** [reply]
It is not in the docs because it was not created and it is not maintained by me, and I discovered it this month.

Have you tried it? Did it work?

I'm open for [docs contributions](https://github.com/navidrome/website) and I think it would be great to add it to the docs, as long as it is mentioned the formula is not maintained by Navidrome project, and point users to the appropriate support channel.

**Reply by Intro1997** [reply]
> It is not in the docs because it was not created and it is not maintained by me, and I discovered it this month.
> 
> Have you tried it? Did it work?
> 
> I'm open for [docs contributions](https://github.com/navidrome/website) and I think it would be great to add it to the docs, as long as it is mentioned the formula is not maintained by Navidrome project, and point users to the appropriate support channel.

Thanks for your reply!

I understand why the method of installing Navidrome via Homebrew wasn't added to the documentation, and I agree with your reasons for not including it.

I've been using Navidrome installed via Homebrew these past few days, and it has been running well without any issues.

Regarding documentation contributions, I'll review the methods and guidelines tomorrow and try to create a PR to add this installation method to the documentation.

**Reply by Intro1997** [reply]
> It is not in the docs because it was not created and it is not maintained by me, and I discovered it this month.
> 
> Have you tried it? Did it work?
> 
> I'm open for [docs contributions](https://github.com/navidrome/website) and I think it would be great to add it to the docs, as long as it is mentioned the formula is not maintained by Navidrome project, and point users to the appropriate support channel.

I found the author of navidrome formula on Homebrew by [blame](https://github.com/Homebrew/homebrew-core/blame/master/Formula/n/navidrome.rb) and created a  [discussion](https://github.com/orgs/Homebrew/discussions/5645)  on Homebrew to ask the two authors about their opinions on adding the Homebrew installation method to the official documentation, which might take some time.

**Reply by Intro1997** [reply]
> It is not in the docs because it was not created and it is not maintained by me, and I discovered it this month.
> 
> Have you tried it? Did it work?
> 
> I'm open for [docs contributions](https://github.com/navidrome/website) and I think it would be great to add it to the docs, as long as it is mentioned the formula is not maintained by Navidrome project, and point users to the appropriate support channel.

Sorry, I'm afraid I can't create a PR to add homebrew installation. [According to the maintainer of homebrew](https://github.com/orgs/Homebrew/discussions/5645#discussioncomment-10799219), the contributor of formula in homebrew has no responsibility to maintain it. If it breaks and nobody is there to fix it it'll just be disabled and removed. I can't add such an unstable installation method to the documentation.

**Reply by deluan** [reply]
Yeah, understandable. Well, it is already listed in the [Community Packages](https://www.navidrome.org/docs/installation/packages/) page. Anyway, after #3125 is merged, I may take a look on how to create a formula with GoReleaser, and may publish it as part of our release.

Thanks for looking into that.
